### DISCUSSION TITLE
WhatsApp

### DISCUSSION THREAD

**Discussion by 0dff03e35e441de3084d705f632e8976** [sepinf-inc/IPED#735] [Ideas]
Very pleasure my name is Thiago and I am a judicial and extrajudicial expert and many of our demands are mobile in whatsapp.  e-mail thiagosp @apecof.org.br

WhatsApp message, decrypted, stop assembling conversations in a friendly way due to chat_list. If necessary I can providea db of whatsapp for testing

There is a preview of when this will go or it can be arranged

==================================================================================

Muito prazer meu nome é Thiago e sou um especialista judicial e extrajudicial e muitas das nossas demandas são móveis no whatsapp.  e-mail thiagosp @apecof.org.br

Mensagem de WhatsApp, descriptografada, pare de montar conversas de forma amigável devido ao chat_list. Se necessário, posso fornecer um db de whatsapp para testes

Há uma prévia de quando isso vai ou pode ser arranjado.

**Comment by lfcnassif** [comment]
Not sure what you are asking here... If whatsapp decoding isn't working, we need a sample WA database to reproduce and fix the possible error.

**Reply by 0dff03e35e441de3084d705f632e8976** [reply]
Em busca da correção de análise do código do avilla


Novo Code

void WhatsappDatabase::getChats(Settings &settings, std::vector<WhatsappChat*> &chats)
{
	const char *query = "SELECT chat_view.raw_string_jid, chat_view.subject, chat_view.created_timestamp, max(messages.timestamp) " \
						"FROM chat_view " \
						"LEFT OUTER JOIN messages on messages.key_remote_jid = chat_view.raw_string_jid " \
						"WHERE chat_view.hidden = '0' "\
						"GROUP BY chat_view.raw_string_jid, chat_view.subject, chat_view.created_timestamp " \
						"ORDER BY max(messages.timestamp) desc";

**Reply by 0dff03e35e441de3084d705f632e8976** [reply]
O string "chat_view" está oculta o código que ela funciona fui passado agora. 
A parte gráfica do WhatsApp , não esta funcional, mas em sql no IPED 3.18.8 eu tenho acesso as conversar.

**Comment by lfcnassif** [comment]
Support for new DB versions was implemented months ago, so closing.
